src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -41,39 +43,70 @@ private RootCommand SetupCommandLineAPI()
             rootCommand = new RootCommand();
             rootCommand.Description = "Analyze Azure Resource Manager (ARM) Templates for security and best practice issues.";
 
-            Option<FileInfo> templateOption = new Option<FileInfo>(
-                    "--template-file-path",
-                    "The ARM template to analyze")
-            {
-                IsRequired = true
-            };
-            templateOption.AddAlias("-t");
-            rootCommand.AddOption(templateOption);
+            // Setup analyze-template w/ template file argument and parameter file option
+            Command analyzeTemplateCommand = new Command(
+                "analyze-template",
+                "Analyze a singe template");
+            
+            Argument<FileInfo> templateArgument = new Argument<FileInfo>(
+                "template-file-path",
+                "The ARM template to analyze");
+            analyzeTemplateCommand.AddArgument(templateArgument);
 
             Option<FileInfo> parameterOption = new Option<FileInfo>(
                  "--parameters-file-path",
                  "The parameter file to use when parsing the specified ARM template");
             parameterOption.AddAlias("-p");
-            rootCommand.AddOption(parameterOption);
+            analyzeTemplateCommand.AddOption(parameterOption);            
+            
+            analyzeTemplateCommand.Handler = CommandHandler.Create<FileInfo, FileInfo>((templateFilePath, parametersFilePath) => this.AnalyzeTemplate(templateFilePath, parametersFilePath));
+
+            // Setup analyze-directory w/ directory argument 
+            Command analyzeDirectoryCommand = new Command(
+                "analyze-directory", 
+                "Analyze all templates within a directory");
+
+            Argument<DirectoryInfo> directoryArgument = new Argument<DirectoryInfo>(
+                "directory-path",
+                "The directory to analyze");
+            analyzeDirectoryCommand.AddArgument(directoryArgument);
 
-            rootCommand.Handler = CommandHandler.Create<FileInfo, FileInfo>((templateFilePath, parametersFilePath) => this.RootCommandHandler(templateFilePath, parametersFilePath));
+            analyzeDirectoryCommand.Handler = CommandHandler.Create<DirectoryInfo>((directoryPath) => this.AnalyzeDirectory(directoryPath));
+
+            // Add commands to root command
+            rootCommand.AddCommand(analyzeTemplateCommand);
```

Comment:
Is there a way to add a helpful message if you try to use the Directory command, but provide the path to a specific file? Right now the output looks something like this `An exception occured: Unable to process directory (<path_to_some_file>.csproj)`

Reply:
Should now say "An exception occured: The parameter is incorrect. : '<path_to_file>'"